### ISSUE TITLE
Separate Custom Settings From Secrets

### ISSUE THREAD

**Issue by tacc-wbomar** [TACC/Core-CMS#39]
# Goal

Extract non-secret custom settings from `secrets.py` so that they can be saved in the repo to reduce the manual maintenance of `secrets.py`.

_Core Portal will do the same for itself via [FP-304]. The "Direction" is based on that of [FP-304]._

# Requirements

1. Non-secret settings __must not__ be in `secrets.py`.
1. Non-secret settings __must__ be comited to the repo.

# Direction

## Portal's Solution

- https://github.com/TACC/Core-Portal/pull/393

## Original Idea

<details><summary>Archived</summary>

1. In the `{env}.env` file (e.g. `dev.env`) of the [TACC/Core-Portal-Deployments](https://github.com/TACC/Core-Portal-Deployments/) repo:

    ```
    CUSTOM_SETTINGS_CMS=dev.custom_settings_cms.py
    ```
1. In the deploy script for the [Jenkins "Core Portal Deploy" job](https://jenkins01.tacc.utexas.edu/job/Core_Portal_Deploy/):

    ```
    cp $CUSTOM_SETTINGS /opt/portal/Camino/conf/portal/custom_settings.py
    ```
3. In [TACC/Core-CMS](https://github.com/TACC/Core-CMS/) `settings.py`:

    ```
    from taccsite_cms.custom_settings import *
    ```
4. Use `custom_settings.py` instead of `.env` for `THEME`.
5. Remove `THEME` from [Jenkins "Core CMS" job](https://jenkins01.tacc.utexas.edu/job/Core_CMS/).
6. Migrate non-secret `settings.py` values to `custom_settings.py`.
7. Remove non-secret `settings.py` values from CMS Stache entries.
8. Update [TACC/Core-Portal](https://github.com/TACC/Core-Portal)'s `sample.secrets.py`.

</details>

# Background

In GitLab by @taoteg on Jul 22, 2020, 18:05

> Right now the branindg configurations for each portal is included in its secrets.py file, but these values are not sensitive (eg - secret).
> 
> In order to better clarify the role of the secrets.py file, and better clarify the location of branding specific configurations, separate all bradning and logo settings out into branding.py and then overwrite the default values (automatically configured using demo values as of the new settings update). This will result in the unique values (only) for each SAD CMS site living in the secrets.py file and overwriting default values living in settings.py, default_settings.py (formerly example_setting.py) and branding.py.

And months later, @rstijerina and @tacc-wbomar [discuss a solution for portal then CMS.](https://tacc-team.slack.com/archives/G0143P1HQAK/p1619034852012300?thread_ts=1619030230.010100&cid=G0143P1HQAK)

[FP-304]: https://jira.tacc.utexas.edu/browse/FP-304

**Comment by tacc-wbomar** [comment]
In GitLab by @tacc-wbomar on Sep 10, 2020, 12:49

__Note__: Issue #37 (branch `task/37-per-site-assets`) identifies the current secret customization per website into a `secrets.py` per website, but does not use them.

**Comment by tacc-wbomar** [comment]
In GitLab by @tacc-wbomar on Nov 19, 2020, 15:07

**Note**: Portal team is also doing this via [FP-304: Separate real secrets from custom settings in settings_secret.py](https://jira.tacc.utexas.edu/browse/FP-304).

**Comment by taoteg** [comment]
Yes, this sounds great. Once this is done, it will be easier to make the case for moving everything into our magical `Core-Projects-Resources` repo to unify the external unique elements.

**Comment by tacc-wbomar** [comment]
P.S. This issue is currently blocking https://github.com/TACC/Core-CMS/pull/192.

**Comment by taoteg** [comment]
https://github.com/TACC/Core-Portal/pull/393 is merged into main on Core-portal, so we can now follow suit here and align the two setups. I am testing the setup in 3dem for the portal container, will replicate here for cms container once I have vetted it. Progress!

**Comment by tacc-wbomar** [comment]
This did not follow the pattern of https://github.com/TACC/Core-Portal/pull/393, but its similar and we hope to update it later this year to match more closely.
